### DISCUSSION TITLE
AttributeError: 'OutStream' object has no attribute 'buffer'

### DISCUSSION THREAD

**Discussion by dataeng1** [openai/whisper#879] [Q&A]
/lib/python3.8/dist-packages/whisper/transcribe.py
introduces the below error which wasn't occurring a couple of weeks ago.
Could you take a look?

Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/gradio/routes.py", line 275, in run_predict
    output = await app.blocks.process_api(
  File "/usr/local/lib/python3.8/dist-packages/gradio/blocks.py", line 760, in process_api
    result = await self.call_function(fn_index, inputs, iterator)
  File "/usr/local/lib/python3.8/dist-packages/gradio/blocks.py", line 671, in call_function
    prediction = await anyio.to_thread.run_sync(
  File "/usr/local/lib/python3.8/dist-packages/anyio/to_thread.py", line 31, in run_sync
    return await get_asynclib().run_sync_in_worker_thread(
  File "/usr/local/lib/python3.8/dist-packages/anyio/_backends/_asyncio.py", line 937, in run_sync_in_worker_thread
    return await future
  File "/usr/local/lib/python3.8/dist-packages/anyio/_backends/_asyncio.py", line 867, in run
    result = context.run(func, *args)
  File "<ipython-input-3-35a38f2d00af>", line 2, in create_textfile
    results = model.transcribe(video_name, verbose=True, language="en")
  File "/usr/local/lib/python3.8/dist-packages/whisper/transcribe.py", line 212, in transcribe
    add_segment(
  **File "/usr/local/lib/python3.8/dist-packages/whisper/transcribe.py", line 172, in add_segment
    sys.stdout.buffer.write(line.encode(sys.getdefaultencoding(), errors="replace"))**
**AttributeError: 'OutStream' object has no attribute 'buffer'**

**Comment by glangford** [comment]
Can you describe the environment you are running in - are you running in a Jupyter notebook or a Colab?

Recent commit
* https://github.com/openai/whisper/commit/9d646db9d885d7d86fa5f40aa29a48147379ed70

may be similar issue:
[Running in Jupyter Notebook raises 'OutStream' object has no attribute 'buffer' #2516](https://github.com/psf/black/issues/2516)

**Reply by glangford** [reply]
I am not a whisper developer, but I would suggest two possible workarounds:
* set `verbose` to False (which bypasses the offending code)
* fall back to an earlier release, such as
https://github.com/openai/whisper/releases/tag/v20230117

cc: @jongwook

**Reply by jongwook** [reply]
Thanks @glangford and @dataeng1, It appears that I was wrong to assume that `sys.stdout.buffer` is always available. I'll probably need to fall back to `print()` or try `os.fdopen` on `sys.stdout.fileno()` instead.

**Reply by dataeng1** [reply]
Thanks, @glangford and @jongwook 
_Can you describe the environment you are running in - are you running in a Jupyter notebook or a Colab?_
It happened when running the below line in Colab.
`model.transcribe(video_name, verbose=True, language="en")

As proposed, changing `verbose=False` suppressed the error occurring by skipping  ` if verbose:` block in transcribe.py, line 172,
however as `verbose=True` is also a useful option to get more information, hopefully, this error gets resolved soon.

**Comment by hchaudhary1** [comment]
17Jan2023 release works without needing to change `verbose`.
```
! pip install git+https://github.com/openai/whisper.git@37a4f1be6d178bc8dad010dc33bcbbc69350e6bf
```

**Reply by ghost** [reply]
Hi @hchaudhary1

I get the following error when I try to install the above... can you advise?

`Collecting git+https://github.com/openai/whisper.git@37a4f1be6d178bc8dad010dc33bcbbc69350e6bf
  Cloning https://github.com/openai/whisper.git (to revision 37a4f1be6d178bc8dad010dc33bcbbc69350e6bf) to c:\users\carl carver\appdata\local\temp\pip-req-build-k5yrzew3
  Running command git clone --filter=blob:none --quiet https://github.com/openai/whisper.git 'C:\Users\Carl Carver\AppData\Local\Temp\pip-req-build-k5yrzew3'
  Running command git rev-parse -q --verify 'sha^37a4f1be6d178bc8dad010dc33bcbbc69350e6bf'
  Running command git fetch -q https://github.com/openai/whisper.git 37a4f1be6d178bc8dad010dc33bcbbc69350e6bf
  Running command git checkout -q 37a4f1be6d178bc8dad010dc33bcbbc69350e6bf
  Resolved https://github.com/openai/whisper.git to commit 37a4f1be6d178bc8dad010dc33bcbbc69350e6bf
  Preparing metadata (setup.py) ... done
Requirement already satisfied: numpy in c:\users\carl carver\appdata\local\programs\python\python311\lib\site-packages (from openai-whisper==20230117) (1.24.1)
**ERROR: Could not find a version that satisfies the requirement torch (from openai-whisper) (from versions: none)
ERROR: No matching distribution found for torch`**

**Reply by hchaudhary1** [reply]
I only tested on Google colab. Local install you can clone & install the exact tag.

**Comment by hanshaoaichiyu** [comment]
I have deal with this problem， this chinese blog maybe can help you ：https://blog.csdn.net/sinat_33353575/article/details/125110025
